File: PYTHON/Grace.py
def	string() -> str:
	s_format = "def	string() -> str:{1}	s_format = {2}{0}{2}{1}	s	= s_format.format(s_format, chr(10), chr(34), chr(39)){1}	return s{1}{1}{1}def	open_and_write(string: str):{1}	with open('Grace_kid.py', 'w+') as file:{1}		file.write(string){1}{1}{1}def	run():{1}	my_string = string(){1}	open_and_write(my_string){1}{1}{1}run(){1}"
	s	= s_format.format(s_format, chr(10), chr(34), chr(39))
	return s


def	open_and_write(string: str):
	with open('Grace_kid.py', 'w+') as file:
		file.write(string)


def	run():
	my_string = string()
	open_and_write(my_string)


run()

File: RUST/Grace.rs
use	std::fs::File;
use	std::io::Write;

/*
 *	Using macros here
 */

macro_rules!	open_and_write_in_file {
	($val:expr) => {
		let	f = File::create("Grace_kid.rs");
		let	_ = f.expect("Error").write($val.as_bytes());
	};
}

macro_rules!	string_to_write {
	($data:expr) => {
		format!("use\tstd::fs::File;\nuse\tstd::io::Write;\n\n/*\n *\tUsing macros here\n */\n\nmacro_rules!\topen_and_write_in_file {{\n\t($val:expr) => {{\n\t\tlet\tf = File::create(\"Grace_kid.rs\");\n\t\tlet\t_ = f.expect(\"Error\").write($val.as_bytes());\n\t}};\n}}\n\nmacro_rules!\tstring_to_write {{\n\t($data:expr) => {{\n\t\tformat!({0:?}, $data)\n\t}};\n}}\n\nmacro_rules!\texec {{\n\t() => {{\n\t\tfn\tmain() {{\n\t\t\topen_and_write_in_file!(string_to_write!({0:?}));\n\t\t}}\n\t}};\n}}\n\nexec!();\n", $data)
	};
}

macro_rules!	exec {
	() => {
		fn	main() {
			open_and_write_in_file!(string_to_write!("use\tstd::fs::File;\nuse\tstd::io::Write;\n\n/*\n *\tUsing macros here\n */\n\nmacro_rules!\topen_and_write_in_file {{\n\t($val:expr) => {{\n\t\tlet\tf = File::create(\"Grace_kid.rs\");\n\t\tlet\t_ = f.expect(\"Error\").write($val.as_bytes());\n\t}};\n}}\n\nmacro_rules!\tstring_to_write {{\n\t($data:expr) => {{\n\t\tformat!({0:?}, $data)\n\t}};\n}}\n\nmacro_rules!\texec {{\n\t() => {{\n\t\tfn\tmain() {{\n\t\t\topen_and_write_in_file!(string_to_write!({0:?}));\n\t\t}}\n\t}};\n}}\n\nexec!();\n"));
		}
	};
}

exec!();
